### DISCUSSION TITLE
How works Yoctopuce Metoe

### DISCUSSION THREAD

**Discussion by beanoccio** [artisan-roaster-scope/artisan#1722] [General]
I have connected a Yoctopuce Meteo to the PC.
It should automatically transfer the ambient temperature, air pressure and humidity to the roasting profile.
Unfortunately, it doesn't work. What's wrong? What's missing?

![image](https://github.com/user-attachments/assets/20f23451-3341-4476-83a0-32ec268a2897)

**Comment by poundy** [comment]
You don't seem to have "Ambient Source" set, it's still showing BT. And have you selected "update"? 
![image](https://github.com/user-attachments/assets/51743ac0-8d4a-439e-a4b5-a06d73d0ea64)
![image](https://github.com/user-attachments/assets/0b30219b-d40e-458e-91e9-898ffd34f601)

**Reply by beanoccio** [reply]
I don't actually know how this function works. I can't find any documentation.
![image](https://github.com/user-attachments/assets/70e3bb4b-a023-476e-9e6e-535151f6eb62)
Artisan shouldn't crash if the user makes incorrect entries. It would be best if the program displayed a message with a reasonable explanation.

**Reply by MAKOMO** [reply]
The crash/hang happens in this case in the Yoctopuce driver which fails to reach out to the device. Nothing can be done on the Artisan side to prevent this nor a message can be displayed, but to remove support for this device. However, if this device is correctly connected things work as they should. They also work as they should if the device is not configured.

There are two ways this works:
1) you configured some Ambient devices
2) you did not configure Ambient devices

In case 1) you can specify an Ambient Source temperature curve which will be averaged on pressing "update" or at the end of the roast to calculate the ambient temperature value. This assumes you have that selected device measuring the ambient temperature during the roast.

In case 2) the ambient information (temperature, humidity, pressure) is taken from the connected ambient device. Again at the end of the roast or if you press "update". Artisan supports some modules from Phidgets and Yoctopuce delivering ambient information.

**Reply by beanoccio** [reply]
Ok, now I know the idea and logic behind this function. Thank you for the explanation.
